packages/ws/src/ws/WebSocketManager.ts

```diff
@@ -178,13 +180,24 @@ export interface OptionalWebSocketManagerOptions {
 	version: string;
 }
 
-export type WebSocketManagerOptions = OptionalWebSocketManagerOptions & RequiredWebSocketManagerOptions;
+export interface WebSocketManagerOptions extends OptionalWebSocketManagerOptions, RequiredWebSocketManagerOptions {}
 
-export type ManagerShardEventsMap = {
```

Comment:
Honestly, I prefer this be kept dynamic over the docs benefit here - esp. considering this isn't that important of a type docs-wise. I've broken things before when it comes to those EE payloads, this removed safety net doesn't help.

Reply:
What safety net? Outside the fact we now need to manually keep these two in sync (which iirc is already needed at the code layer)

Also this type can definitely be used for documentation purposes, even if it's not crystal clear what it represents (although most can _hopefully_ infer that its `eventName: [event argument]`

Reply:
Doesn't AEE make it so events not in the map type can just be fired with `any` args? If so, the sync isn't type-required at the code level, this is also trivial to miss in code review.

Reply:
I don't think I allow emitting random events, but tbf I haven't tested that. Lets see

Reply:
![Code - Insiders - 2023-12-10 at 20 53 27](https://github.com/discordjs/discord.js/assets/17960496/dd9f366c-2c70-4e60-99aa-ce312640a31e)

So, yeah, shouldn't be an issue

Reply:
sure, whatever then

Reply:
Added a type test to ensure those two event maps keep in sync, as suggested by @almeidx .